Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,687 @@
+    network_ref[i].ack_base = network_ref[i].ack_num;
+    network_ref[i].seq_num = 0x1000000;
+    network_ref[i].window_size = Common::swap16(tcpdata->window_size);
+    network_ref[i].type = IPPROTO_TCP;
+    for (auto& tcp_buf : network_ref[i].tcp_buffers)
+      tcp_buf.used = false;
+    network_ref[i].from.sin_addr.s_addr = *(u32*)&ipdata->destination_addr;
+    network_ref[i].from.sin_port = tcpdata->destination_port;
+    network_ref[i].to.sin_addr.s_addr = *(u32*)&ipdata->source_addr;
+    network_ref[i].to.sin_port = tcpdata->source_port;
+    memcpy(&network_ref[i].bba_mac, &m_eth_ref->mBbaMem[BBA_NAFR_PAR0], Common::MAC_ADDRESS_SIZE);
+    network_ref[i].my_mac = fake_mac;
+    network_ref[i].tcp_socket.setBlocking(false);
+
+    // reply with a sin_ack
+    memset(&m_in_frame[0], 0, 0x100);
+    Common::EthernetHeader* hwpart = (Common::EthernetHeader*)&m_in_frame[0];
+    Common::IPv4Header* ippart = (Common::IPv4Header*)&m_in_frame[14];
+    Common::TCPHeader* tcppart = (Common::TCPHeader*)&m_in_frame[0x22];
+
+    *hwpart = Common::EthernetHeader(IP_PROTOCOL);
+    hwpart->destination = network_ref[i].bba_mac;
+    hwpart->source = fake_mac;
+
+    *ippart = Common::IPv4Header(28, IPPROTO_TCP, network_ref[i].from, network_ref[i].to);
+
+    *tcppart = Common::TCPHeader(network_ref[i].from, network_ref[i].to, network_ref[i].seq_num,
+                                 network_ref[i].ack_num, 0x70 | TCP_FLAG_SIN | TCP_FLAG_ACK);
+    u8 options[] = {0x02, 0x04, 0x05, 0xb4, 0x01, 0x01, 0x01, 0x01};
+    memcpy(&m_in_frame[0x36], options, std::size(options));
+
+    // do checksum
+    tcppart->checksum = Common::ComputeTCPNetworkChecksum(network_ref[i].from, network_ref[i].to,
+                                                          tcppart, 28, IPPROTO_TCP);
+
+    network_ref[i].seq_num++;
+    target = sf::IpAddress(Common::swap32(ipdata->destination_addr));
+    network_ref[i].tcp_socket.connect(target, Common::swap16(tcpdata->destination_port));
+    network_ref[i].ready = false;
+    network_ref[i].ip = *(u32*)ipdata->destination_addr;
+
+    memcpy(&network_ref[i].tcp_buffers[0].data, &m_in_frame[0], 0x3e);
+    network_ref[i].tcp_buffers[0].data_size = 0x3e;
+    network_ref[i].tcp_buffers[0].seq_id = network_ref[i].seq_num - 1;
+    network_ref[i].tcp_buffers[0].tick = GetTickCountStd() - 1900;  // delay
+    network_ref[i].tcp_buffers[0].used = true;
+  }
+  else
+  {
+    // data packet
+    int i = GetTCPSlot(tcpdata->source_port, tcpdata->destination_port,
+                       *(u32*)ipdata->destination_addr);
+    if (i == -1)
+      return;  // not found
+
+    int c = (tcpdata->properties & 0xf0) >> 2;  // header size
+    int size = Common::swap16(ipdata->total_len) - 20 - c;
+    u32 this_seq = Common::swap32(tcpdata->sequence_number);
+
+    if (size > 0)
+    {
+      // only if data
+      if ((int)(this_seq - network_ref[i].ack_num) >= 0)
```

Comment:
Can't the following condition be used instead?
```C++
if (this_seq >= network_ref[i].ack_num)
```

Reply:
No, the seq number can roll around and then 00000010 is still > FFFFFFFF in the flow